### PR TITLE
NavigationState handling

### PR DESCRIPTION
**Pull request by alexanderbock**
Change the navigation state format from Lua to JSON (applies to `openspace.navigation.saveNavigationState` and `openspace.navigation.loadNavigationState`. Before, the `yaw` and `pitch` values were ignored if the `up` vector was not specified. Now, these three values are loaded independently.

Add the ability to set the navigation state in the profile editor based on the new JSON format.

### CODE REVIEW

**Comment by alexanderbock** [comment]
Closes #2117 when merged

File: src/navigation/navigationhandler.cpp [hunk 1]
```diff
@@ -578,22 +578,15 @@ void NavigationHandler::loadNavigationState(const std::string& filepath) {
...
-            "NavigationState"
-        );
-        setNavigationStateNextFrame(NavigationState(navigationStateDictionary));
-    }
-    catch (ghoul::RuntimeError& e) {
-        LERROR(fmt::format("Unable to set camera position: {}", e.message));
-    }
+    std::ifstream f(filepath);
+    std::string contents = std::string(
+        (std::istreambuf_iterator<char>(f)),
```
Comment (WeirdRubberDuck) [inline src/navigation/navigationhandler.cpp:583]:
```suggestion
        std::istreambuf_iterator<char>(f),
```
?

File: src/navigation/navigationstate.cpp [hunk 2]
```diff
@@ -89,11 +89,39 @@ NavigationState::NavigationState(const ghoul::Dictionary& dictionary) {
...
+
+NavigationState::NavigationState(const nlohmann::json& json) {
+    position.x = json["position"]["x"].get<double>();
+    position.y = json["position"]["y"].get<double>();
+    position.z = json["position"]["z"].get<double>();
+
+    anchor = json["anchor"];
+
+    if (auto it = json.find("referenceframe");  it != json.end()) {
+        referenceFrame = it->get<std::string>();
```
Comment (WeirdRubberDuck) [inline src/navigation/navigationstate.cpp:105]: This should be set to the anchor node if the reference frame was not specified

Comment (WeirdRubberDuck) [reply src/navigation/navigationstate.cpp:105]: Currently, loading a navigation stat ethat looks like this leads to an error:
```
{
  "anchor": "Earth",
  "position": {
    "x": 13514805.101321544,
    "y": 5454659.039307959,
    "z": -18279333.341204002
  },
  "up": {
    "x": 0.8035739201794957,
    "y": 0.0035944893626913277,
    "z": 0.59519411493527
  }
}
```

image

**Review by WeirdRubberDuck — CHANGES_REQUESTED** [review]
Noticed that it is still not saved using a file extension.  Should we?  

Either json, or maybe our custom one 

Also, note to us: this is a highly breaking change and should be included in the list of breaking changes that I have started adding to the docs in this PR: 

https://github.com/OpenSpace/OpenSpace-Docs/pull/11  
(still waiting for someone to read it)

**Comment by alexanderbock** [comment]
If the path provided to the `saveNavigationState` doesn't have an extension, a `.navstate` one is now added. If the user _did_ provide a different extension, that one is used instead

**Pull request merged**